### PR TITLE
Limit several project settings

### PR DESCRIPTION
**Pull request by Superwaitsum**
As seen in https://github.com/godotengine/godot/issues/21795 , we need to limit some project settings.
I will be adding more limited settings later this week, but I'm opening this PR early for reviewing of the values I'm using.

Fixes https://github.com/godotengine/godot/issues/21795

### CODE REVIEW

**Comment by akien-mga** [comment]
Looks fine so far.

**Comment by Superwaitsum** [comment]
Good, now I pushed the last batch of limited settings, please tell me if they're ok to squash for merging.

File: core/io/file_access_network.cpp [hunk 1]
```diff
@@ -500,8 +500,11 @@ uint64_t FileAccessNetwork::_get_modified_time(const String &p_file) {
 void FileAccessNetwork::configure() {
 
 	GLOBAL_DEF("network/remote_fs/page_size", 65536);
+	ProjectSettings::get_singleton()->set_custom_property_info("network/remote_fs/page_size", PropertyInfo(Variant::INT, "network/remote_fs/page_size", PROPERTY_HINT_RANGE, "1, 65536, 1, or_greater")); //is used as denominator and can't be zero
```
Comment (akien-mga) [inline core/io/file_access_network.cpp]: The max doesn't give much room for increasing the page size, given that the default value is already 65536. Maybe it should be a bit more?

File: core/io/file_access_network.cpp [hunk 2]
```diff
@@ -500,8 +500,11 @@ uint64_t FileAccessNetwork::_get_modified_time(const String &p_file) {
...
 	GLOBAL_DEF("network/remote_fs/page_read_ahead", 4);
+	ProjectSettings::get_singleton()->set_custom_property_info("network/remote_fs/page_read_ahead", PropertyInfo(Variant::INT, "network/remote_fs/page_read_ahead", PROPERTY_HINT_RANGE, "0, 8, 1, or_greater"));
 	GLOBAL_DEF("network/remote_fs/max_pages", 20);
+	ProjectSettings::get_singleton()->set_custom_property_info("network/remote_fs/max_pages", PropertyInfo(Variant::INT, "network/remote_fs/max_pages", PROPERTY_HINT_RANGE, "0, 40, 1, or_greater")); //unused property anywhere
```
Comment (akien-mga) [inline core/io/file_access_network.cpp]: Please remove it then.

File: core/io/file_access_network.cpp [hunk 3]
```diff
@@ -500,8 +500,11 @@ uint64_t FileAccessNetwork::_get_modified_time(const String &p_file) {
 void FileAccessNetwork::configure() {
 
 	GLOBAL_DEF("network/remote_fs/page_size", 65536);
+	ProjectSettings::get_singleton()->set_custom_property_info("network/remote_fs/page_size", PropertyInfo(Variant::INT, "network/remote_fs/page_size", PROPERTY_HINT_RANGE, "1, 65536, 1, or_greater")); //is used as denominator and can't be zero
 	GLOBAL_DEF("network/remote_fs/page_read_ahead", 4);
+	ProjectSettings::get_singleton()->set_custom_property_info("network/remote_fs/page_read_ahead", PropertyInfo(Variant::INT, "network/remote_fs/page_read_ahead", PROPERTY_HINT_RANGE, "0, 8, 1, or_greater"));
```
Comment (akien-mga) [inline core/io/file_access_network.cpp]: We don't put a space after commas for hint range strings currently. It's not a strong convention but it's the predominant usage, so it's best to stick to it until we decide on a different convention.

File: main/main.cpp [hunk 4]
```diff
@@ -1641,7 +1649,7 @@ bool Main::start() {
 			GLOBAL_DEF("display/window/stretch/aspect", "ignore");
 			ProjectSettings::get_singleton()->set_custom_property_info("display/window/stretch/aspect", PropertyInfo(Variant::STRING, "display/window/stretch/aspect", PROPERTY_HINT_ENUM, "ignore,keep,keep_width,keep_height,expand"));
 			GLOBAL_DEF("display/window/stretch/shrink", 1);
-			ProjectSettings::get_singleton()->set_custom_property_info("display/window/stretch/shrink", PropertyInfo(Variant::STRING, "display/window/stretch/shrink", PROPERTY_HINT_RANGE, "1,8,1"));
+			ProjectSettings::get_singleton()->set_custom_property_info("display/window/stretch/shrink", PropertyInfo(Variant::REAL, "display/window/stretch/shrink", PROPERTY_HINT_RANGE, "1,8,1")); //was using string, not sure if values need changing
```
Comment (akien-mga) [inline main/main.cpp]: No need for this comment, the fix is correct.

**Comment by gcardozo123** [comment]
Does this PR fixes all ranges that need to be limited in the Project Setting? If so, I'd consider assigning 	#21795 to @Superwaitsum.

**Comment by Superwaitsum** [comment]
@gcardozo123 this PR fixes all ranges needed, but I think assigning only works on members of the organization, so I can't be assigned.

**Comment by akien-mga** [comment]
Looks all good to me, once squashed I'll merge.

**Comment by Superwaitsum** [comment]
@akien-mga Squashed and ready to merge

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**